runtime/src/fees.rs

```diff
@@ -111,9 +113,11 @@ impl<T: Trait> Module<T> {
         }
     }
 
-    fn can_change_fee(_who: T::AccountId) -> DispatchResult {
-        //TODO add auth who can change fees
-        //        ensure!(<validatorset::Module<T>>::is_validator(who), "Not authorized to change fees.");
+    fn can_change_fee(origin: T::Origin) -> DispatchResult {
+        T::FeeChangeOrigin::try_origin(origin)
+            .map(|_| ())
+            .or_else(ensure_root)?;
```

Comment:
we fallback here to check if the fee is changed by the root ?

Reply:
Exactly. If `try_origin` fails, it will return an `Err(OuterOrigin)` result. The `or_else` will unwrap the `OuterOrigin` and use it as an argument to `ensure_root`.